File: Admin_p/Admin_p.Web/Modules/electroDB/PromotionProducts/PromotionProductsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Admin_p.electroDB.Columns;

[ColumnsScript("electroDB.PromotionProducts")]
[BasedOnRow(typeof(PromotionProductsRow), CheckNames = true)]
public class PromotionProductsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int PromotionProductId { get; set; }
    public string PromotionName { get; set; }
    public string ProductName { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/PromotionProductsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PromotionProductsRow } from "./PromotionProductsRow";

export interface PromotionProductsColumns {
    PromotionProductId: Column<PromotionProductsRow>;
    PromotionName: Column<PromotionProductsRow>;
    ProductName: Column<PromotionProductsRow>;
    CreatedAt: Column<PromotionProductsRow>;
    UpdatedAt: Column<PromotionProductsRow>;
}

export class PromotionProductsColumns extends ColumnsBase<PromotionProductsRow> {
    static readonly columnsKey = 'electroDB.PromotionProducts';
    static readonly Fields = fieldsProxy<PromotionProductsColumns>();
}